robot_smach_states/src/robot_smach_states/utility.py

```diff
@@ -357,22 +357,25 @@ class ResolveArm(smach.State):
     def __init__(self, arm, state_machine):
         """ Resolves, if possible, an arm for a state machine taking into account all the arm requirements
 
-        :param arm: given arm designator
+        :param arm: (lockable) arm designator
         :param state_machine: used state machine
         """
         smach.State.__init__(self, outcomes=['succeeded', 'failed'])
+
+        if not isinstance(arm, LockingDesignator):
```

Comment:
I don't think this will work on the higher leve. As the original designator, `arm` is still used by the other states. So those aren't locked, but these should be! So we should check indeed that `arm` is a `LockingDesignator`, but we should raise an error. Not create a `LockingDesignator` ourselves